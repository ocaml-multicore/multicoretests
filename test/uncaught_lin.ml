(* Test of the behaviour of Lin tests with uncaught exceptions *)

let always_fail () = failwith "unexpected"

module UncaughtExcConf = struct
  open Lin

  type t = unit

  let init () = ()
  let cleanup _ = ()
  let api = [ val_ "always_fail" always_fail (t @-> returning unit) ]
end

module UncaughtExc_domain = Lin_domain.Make (UncaughtExcConf)

let _ =
  QCheck_base_runner.run_tests_main
    [
      UncaughtExc_domain.lin_test ~count:10
        ~name:"Lin DSL test of uncaught exceptions";
      UncaughtExc_domain.neg_lin_test ~count:10
        ~name:"neg Lin DSL test of uncaught exceptions";
    ]
