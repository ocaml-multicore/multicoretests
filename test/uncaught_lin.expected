random seed: 260395858
generated error fail pass / total     time test name

=== Error ======================================================================

Test Lin DSL test of uncaught exceptions errored on:

                           |             
                     always_fail t       
                           |             
                .---------------------.
                |                     |             


exception Failure("unexpected") raised but not caught while running always_fail t


=== Error ======================================================================

Test neg Lin DSL test of uncaught exceptions errored on:

                           |             
                     always_fail t       
                           |             
                .---------------------.
                |                     |             


exception Failure("unexpected") raised but not caught while running always_fail t

================================================================================
failure (0 tests failed, 2 tests errored, ran 2 tests)
